File: tmdb-core/src/main/java/dev/reuss/tmdb/domain/movie/model/MovieTranslations.java
package dev.reuss.tmdb.domain.movie.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import dev.reuss.tmdb.common.TmdbModel;
import dev.reuss.tmdb.common.translation.Translation;
import dev.reuss.tmdb.common.translation.TranslationsResponse;
import dev.reuss.tmdb.util.TmdbCollections;

import java.util.List;

/**
 * Translations for a TMDB movie.
 *
 * @param id           TMDB movie id
 * @param translations movie translations
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record MovieTranslations(
        int id,
        List<Translation<MovieTranslationData>> translations
) implements TranslationsResponse<MovieTranslationData>, TmdbModel {

    public MovieTranslations {
        translations = TmdbCollections.list(translations);
    }
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/domain/movie/model/MovieTranslations.kt
package dev.reuss.tmdb.domain.movie.model

import com.fasterxml.jackson.annotation.JsonProperty

import dev.reuss.tmdb.common.TmdbModel
import dev.reuss.tmdb.common.translation.Translation
import dev.reuss.tmdb.common.translation.TranslationsResponse

/**
 * Translations for a TMDB movie.
 *
 * @property id TMDB movie id
 * @property translations movie translations
 */
@JvmRecord
data class MovieTranslations(
    @all:JsonProperty("id")
    val id: Int,

    @all:JsonProperty("translations")
    override val translations: List<Translation<MovieTranslationData>> = emptyList()
) : TranslationsResponse<MovieTranslationData>, TmdbModel
